towhee/engine/operator_pool.py

```diff
@@ -13,23 +13,56 @@
 # limitations under the License.
 
 
-from towhee.operator.operator import Operator
+from towhee.operator.base import OperatorBase
+from towhee.engine.operator_loader import OperatorLoader
+from towhee.engine.task import Task
 
 
 class OperatorPool:
-    """
-    OperatorPool manages Operator creation, acquisition, release, garbage collection.
-    Each TaskExecutor has one OperatorPool.
+    """`OperatorPool` manages `Operator` creation, acquisition, release, and garbage
+    collection. Each `TaskExecutor` has one `OperatorPool`.
     """
 
-    def acquire(self, name: str, init_args: dict) -> Operator:
-        """
-        Acquire an Operator by name.
-        """
-        raise NotImplementedError
+    def __init__(self, cache_path: str = None):
+        self._op_loader = OperatorLoader(cache_path)
+        self._all_ops = {}
+
+    @property
+    def available_ops(self):
+        return self._all_ops.keys()
+
+    def acquire_op(self, task: Task) -> OperatorBase:
+        """Given a `Task`, instruct the `OperatorPool` to reserve and return the
+        specified operator for use in the executor.
 
-    def release(self, op: Operator) -> None:
+        Args:
+            name: (`str`)
+                Unique operator name, as specified in the graph.
+            args: (`dict`)
+                The operator's initialization arguments, if any.
+
+        Returns:
+            (`towhee.OperatorBase`)
+                The operator instance reserved for the caller.
         """
-        Release an Operator.
+        if task.op_name not in self._all_ops:
+            op = self._op_loader.load_operator(task.op_func, task.op_args)
+            op.name = task.op_name
+        else:
```

Comment:
Use op_name and init params to calc a key, because the same operator with different init params is a different object.

Reply:
Good catch, I'll fix this.